File: Python/15681.py
import sys;input=sys.stdin.readline

def makeTree(currentNode, parent):
    for Node in connect[currentNode]:
        if Node != parent:
            currentNodeChild[currentNode].append(Node)
            makeTree(Node, currentNode)

def countSubtreeNodes(currentNode):
    size[currentNode] = 1
    for Node in currentNodeChild[currentNode]:
        countSubtreeNodes(Node)
        size[currentNode] += size[Node]

N,R,Q = map(int, input().split())
connect = [[] for _ in range(N+1)]
currentNodeChild = [[] for _ in range(N+1)]
size = [0] * (N+1)

for _ in range(N-1):
    U,V = map(int, input().split())
    connect[U].append(V)
    connect[V].append(U)

makeTree(R, -1)
countSubtreeNodes(R)

for _ in range(Q):
    print(size[int(input())])

File: C++/15681.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<vector<int>> connect, currentNodeChild;
vector<int> size;

void makeTree(int currentNode, int parent) {
    for(int Node : connect[currentNode]) {
        if (Node != parent) {
            currentNodeChild[currentNode].push_back(Node);
            makeTree(Node, currentNode);
        }
    }
}

void countSubtreeNodes(int currentNode) {
    size[currentNode] = 1;
    for(int Node : currentNodeChild[currentNode]) {
        countSubtreeNodes(Node);
        size[currentNode] += size[Node];
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N,R,Q,U,V;
    cin >> N >> R >> Q;

    connect.resize(N + 1);
    currentNodeChild.resize(N + 1);
    size.resize(N + 1, 0);

    for(int i=0; i<N-1; i++) {
        cin >> U >> V;
        connect[U].push_back(V);
        connect[V].push_back(U);
    }

    makeTree(R, -1);
    countSubtreeNodes(R);

    for (int i = 0; i < Q; ++i){
        int query;
        cin >> query;
        cout << size[query] << "\n";
    }

    return 0;
}